p2p/src/network/pubsub/p2p_network_pubsub_reducer.rs

```diff
@@ -227,6 +107,137 @@ impl P2pNetworkPubsubState {
                         dbg!(err);
                     }
                 }
+
+                for subscription in &subscriptions {
+                    let topic_id = subscription.topic_id().to_owned();
+                    let topic = self.topics.entry(topic_id).or_default();
+
+                    if subscription.subscribe() {
+                        if let Entry::Vacant(v) = topic.entry(*peer_id) {
+                            v.insert(P2pNetworkPubsubClientTopicState::default());
+                        }
+                    } else {
+                        topic.remove(peer_id);
+                    }
+                }
+
+                for graft in &control.graft {
+                    if let Some(mesh_state) = self
+                        .topics
+                        .get_mut(graft.topic_id())
+                        .and_then(|m| m.get_mut(peer_id))
+                    {
+                        mesh_state.mesh = P2pNetworkPubsubClientMeshAddingState::Added;
+                    }
+                }
+                for prune in &control.prune {
+                    if let Some(mesh_state) = self
+                        .topics
+                        .get_mut(prune.topic_id())
+                        .and_then(|m| m.get_mut(peer_id))
+                    {
+                        mesh_state.mesh = P2pNetworkPubsubClientMeshAddingState::TheyRefused;
+                    }
+                }
+                for iwant in &control.iwant {
+                    for msg_id in &iwant.message_ids {
+                        if let Some(msg) = self.mcache.map.get(msg_id) {
+                            if let Some(client) = self.clients.get_mut(peer_id) {
+                                client.message.publish.push(msg.clone());
+                            }
+                        }
+                    }
+                }
+                for ihave in &control.ihave {
+                    let message_ids = ihave
+                        .message_ids
+                        .iter()
+                        .filter(|msg_id| !self.mcache.map.contains_key(*msg_id))
+                        .cloned()
+                        .collect();
+                    if let Some(client) = self.clients.get_mut(peer_id) {
+                        let ctr = client.message.control.get_or_insert_with(Default::default);
+                        ctr.iwant.push(pb::ControlIWant { message_ids })
+                    }
+                }
+            }
+            P2pNetworkPubsubAction::IncomingMessage {
+                peer_id,
+                message,
+                seen_limit,
+            } => {
+                self.incoming_transactions.clear();
+                self.incoming_snarks.clear();
+
+                let Some(state) = self.clients.get_mut(peer_id) else {
+                    return;
```

Comment:
Is this normal or an error condition?

Reply:
I mean, "normal" as in, the pattern is one we have everywhere, but it should never return here